### ISSUE TITLE
[Buildbot][IOS] Fix for deploying via AltStore/SideStore

### ISSUE THREAD

**Issue by loial** [libretro/RetroArch#15828]
TL;DR Small fix in https://buildbot.libretro.com/stable/altstore.json would vastly improve ability for iOS users to install and update RetroArch by using AltStore/SideStore to sideload.

I couldn't figure out how to make a PR of this (since I couldn't find where 'altstore.json' would be mentioned in source or ci scripts, maybe was manually uploaded?), or most fitting thread/channel in forum/discord, so I'll just file this generic issue, and try to tag @warmenhoven , which seems most active contributer on the ios part lately.

I've been using SideStore (a fork of AltStore) which most importantly lets iOS users sideload apps with only their iPhone (after first setup, no need for computer ot cable, at all), and secondly allowing custom Appstores.

I was very happy to find out someone (Warmenhoven?) had placed an altstore.json on tthe buildbot server (as I've had to create and maintain my own), but Sidestore chokes when trying to load it.

The only missing part is that the app object in the 'apps' array needs its own 'downloadURL' key, basicly a duplicate of the 'downloadURL' key from the current version object in the app's 'versions' array.

I just tested this by proxying the json on my local server, running a simple filter on it ('.apps[0].downloadURL=.apps[0].versions[0].downloadURL') and it loaded perfectly in SideStore.

This should probably be implemented in the automatic build, (maybe even accomodating several versions, hey maybe even betas) but I could attach my fixed version for quick upload.

**Comment by warmenhoven** [comment]
Can you please confirm that https://buildbot.libretro.com/nightly/apple/ios-arm64/altstore.json works for you? (It's for the nightly build, which actually builds more often than nightly.)

**Comment by loial** [comment]
Yeah, The nightly json works fine :)

Though I haven’t tried setting up betas myself, it seems like the altstore.json format supports that, maybe as an alternate install in the client. If not in a single combined json (seems beta:true must be set on app level, not version level?), then maybe set beta flag in the nightly json.

**Comment by warmenhoven** [comment]
Great, thanks for confirming! The same script that updates the nightly altstore.json also creates the one for the release build, so should be good for future releases. I can look into manually updating the release altstore.json.

Regarding setting the beta flag, I'm a little hesitant to combine release and nightly into one file. The "nightly" build is actually built a few times a day and because all of the cores are packaged in the IPA it's a significant download, and I'm not sure I really want to be advertising it to people that wouldn't normally be inclined to go look for it. It sounds like it's not really supported anyway? In which case what's the point of the beta flag? A quick look a the code makes it look like it just changes the badging slightly? I'd prefer to keep the release and nightly script/json file in sync as much as possible to make it easier to debug and test issues like this one. If there's a more compelling reason to set the beta flag than just a badge I'm willing to consider it.
